lib/utils/flock/flock.go

```diff
@@ -0,0 +1,75 @@
+//go:build !windows
+
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package flock
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"syscall"
+
+	"github.com/gravitational/trace"
+)
+
+// Lock implements filesystem lock for blocking another process execution until this lock is released.
+func Lock(ctx context.Context, lockFile string) (func(), error) {
```

Comment:
Could we use the implementation here? https://github.com/gravitational/teleport/pull/46418/files#diff-d099d69928110c77b90c52fa55c9d973aa6070b7f1653c3eb6f6cd3c65352643R314-R350

It addresses a few issues pointed out below, and is easier to test due to LOCK_NB flag option.